File: app/src/main/java/com/sortedqueue/programmercreek/fragments/AllProgramsFragment.java
package com.sortedqueue.programmercreek.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.sortedqueue.programmercreek.R;
import com.sortedqueue.programmercreek.adapter.AllProgramsPagerAdapter;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by Alok on 08/08/17.
 */

public class AllProgramsFragment extends Fragment {

    @BindView(R.id.allProgramsTabLayout)
    TabLayout allProgramsTabLayout;
    @BindView(R.id.programsViewPager)
    ViewPager programsViewPager;
    Unbinder unbinder;

    private static AllProgramsFragment allProgramsFragment;
    public static AllProgramsFragment getInstance() {
        if( allProgramsFragment == null ) allProgramsFragment = new AllProgramsFragment();
        return allProgramsFragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View fragmentView = inflater.inflate(R.layout.fragment_all_programs, container, false);
        unbinder = ButterKnife.bind(this, fragmentView);
        return fragmentView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ArrayList<Fragment> userProgramFragments = new ArrayList<>();
        userProgramFragments.add( UserProgramsFragment.newInstance("C"));
        userProgramFragments.add( UserProgramsFragment.newInstance("C++"));
        userProgramFragments.add( UserProgramsFragment.newInstance("Java"));
        programsViewPager.setAdapter(new AllProgramsPagerAdapter(getChildFragmentManager(), userProgramFragments));
        allProgramsTabLayout.setupWithViewPager(programsViewPager);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/fragments/AllProgramsFragment.kt
package com.sortedqueue.programmercreek.fragments

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.sortedqueue.programmercreek.R
import com.sortedqueue.programmercreek.adapter.AllProgramsPagerAdapter

import java.util.ArrayList




import kotlinx.android.synthetic.main.fragment_all_programs.*

/**
 * Created by Alok on 08/08/17.
 */

class AllProgramsFragment : Fragment() {


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val fragmentView = inflater!!.inflate(R.layout.fragment_all_programs, container, false)
        return fragmentView
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val userProgramFragments = ArrayList<Fragment>()
        userProgramFragments.add(UserProgramsFragment.newInstance("C"))
        userProgramFragments.add(UserProgramsFragment.newInstance("C++"))
        userProgramFragments.add(UserProgramsFragment.newInstance("Java"))
        programsViewPager!!.adapter = AllProgramsPagerAdapter(childFragmentManager, userProgramFragments)
        allProgramsTabLayout!!.setupWithViewPager(programsViewPager)
    }

    override fun onDestroyView() {
        super.onDestroyView()

    }

    companion object {

        private var allProgramsFragment: AllProgramsFragment? = null
        val instance: AllProgramsFragment
            get() {
                if (allProgramsFragment == null) allProgramsFragment = AllProgramsFragment()
                return allProgramsFragment!!
            }
    }
}
